spec/features/users/sign_in_spec.rb

```diff
@@ -91,6 +91,21 @@
     end
   end
 
+  context 'user only signs in via email and password', js: true do
+    it 'displays the session timeout warning' do
+      allow(Figaro.env).to receive(:session_check_frequency).and_return('1')
+      allow(Figaro.env).to receive(:session_check_delay).and_return('2')
+      allow(Figaro.env).to receive(:session_timeout_warning_seconds).
+        and_return(Devise.timeout_in.to_s)
+
+      user = create(:user, :signed_up)
+      sign_in_user(user)
+      visit user_two_factor_authentication_path
+
+      expect(page).to have_css('#session-timeout-msg')
```

Comment:
Yes, i agree! I was conflating that modal with a user being logged in, not just having an active session. Your solution is definitely cleaner since it uses the same logic we had and lets the framework render the correct partial based on the data it's provided.